exercises/binary-search-tree/binary_search_tree_test.go

```diff
@@ -45,8 +45,12 @@ func TestInsertingLess(t *testing.T) {
 		t.Errorf("bst.data: %d, want %d.", actual, expected)
 	}
 
-	actual = bst.left.data
 	expected = 2
+	if bst.left == nil {
+		t.Errorf("bst.left: nil, want %d.", expected)
```

Comment:
could consider using `t.Fatalf` instead of `t.Errorf` + `return` (applies to other lines too but I don't want to repeat myself)